parser.cpp

```diff
@@ -193,24 +197,18 @@ class Parser {
                 if (*(str + 1) == '3' && *(str + 2) == '2') {
                     advance_pos(data, 3);
 
-                    if (parse_rest_of_identifier(data) == token::IDENTIFIER) {
-                        return token::IDENTIFIER;
-                    }
-
-                    // if next char is not part of an identifier, we found a u32 type
-                    return token::U32_TYPE;
+                    if (!(isalnum(**data) || **data == '_'))
```

Comment:
Make a helper function for this: `is_identifier_char`.

Reply:
Also: Was there something that didn't parse if we didn't switch the condition around?

Reply:
yes, U32 was getting parsed as IDENTIFIER earlier

Reply:
Oops. Thanks for fixing this!